src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -3324,14 +3326,33 @@ bool BlockchainLMDB::get_blocks_from(uint64_t start_height, size_t min_block_cou
         throw0(DB_ERROR(lmdb_error("Error attempting to retrieve transaction data from the db: ", result).c_str()));
       tx_blob.assign((const char*)v.mv_data, v.mv_size);
 
-      if (!pruned)
-      {
+      if (pruned) {
+        // get the prunable hash
+
+        // Look up each transaction's tx_id
+        MDB_val_set(tx_hash_val, tx_hash);
+        result = mdb_cursor_get(m_cur_tx_indices, (MDB_val *)&zerokval, &tx_hash_val, MDB_GET_BOTH);
+        if (result)
+          throw0(DB_ERROR(lmdb_error("Error retrieving tx index for prunable hash: ", result).c_str()));
+
+        const txindex *tip = (const txindex *)tx_hash_val.mv_data;
+        MDB_val_set(val_tx_id_for_hash, tip->data.tx_id);
+
+        result = mdb_cursor_get(m_cur_txs_prunable_hash, &val_tx_id_for_hash, &v, MDB_SET);
+        if (result)
+          throw0(DB_ERROR(lmdb_error("Error attempting to retrieve transaction data from the db: ", result).c_str()));
+
+        crypto::hash prunable_hash = *(const crypto::hash*)v.mv_data;
+        current_block.second.push_back(std::make_pair(prunable_hash, std::move(tx_blob)));
```

Comment:
NACK replacing the TXID with the prunable hash, this silently breaks the API of `BlockchainDB`. If we must update the API here, the function signature should be a breaking change so that downstream fails to compile. That, or we can write a new `get_blocks_from()` method override and implement the older override with the newer override so that it's non-breaking.